com.unity.ml-agents/Runtime/Sensors/StackingSensor.cs

```diff
@@ -128,17 +147,42 @@ public string GetName()
         }
 
         /// <inheritdoc/>
-        public virtual byte[] GetCompressedObservation()
+        public byte[] GetCompressedObservation()
         {
-            return null;
+            var compressed = m_WrappedSensor.GetCompressedObservation();
+            m_StackedCompressedObservations[m_CurrentIndex] = compressed;
+
+            int bytesLength = 0;
+            foreach (byte[] compressedObs in m_StackedCompressedObservations)
+            {
+                bytesLength += compressedObs.Length;
+            }
+
+            byte[] bytes = new byte[bytesLength];
+            int offset = 0;
+            for (var i = 0; i < m_NumStackedObservations; i++)
+            {
+                var obsIndex = (m_CurrentIndex + 1 + i) % m_NumStackedObservations;
+                Buffer.BlockCopy(m_StackedCompressedObservations[obsIndex],
+                    0, bytes, offset, m_StackedCompressedObservations[obsIndex].Length);
+                offset += m_StackedCompressedObservations[obsIndex].Length;
+            }
+
+            return bytes;
         }
 
         /// <inheritdoc/>
-        public virtual SensorCompressionType GetCompressionType()
+        public SensorCompressionType GetCompressionType()
         {
-            return SensorCompressionType.None;
+            return m_WrappedSensor.GetCompressionType();
         }
 
-        // TODO support stacked compressed observations (byte stream)
+        public byte[] CreateEmptyPNG()
+        {
+            int height = m_WrappedSensor.GetObservationShape()[0];
```

Comment:
This is fine for now, but it would be even better if we could write a PNG directly instead of creating the empty Texture. That can be a later optimization though.

Reply:
I was thinking of doing that but didn't find a way to directly create an empty PNG file. Do you have any solutions in mind?

Reply:
No, I looked into it a bit and it might be non-trivial. It can definitely happen later.